ESP8266Interface.cpp

```diff
@@ -64,20 +67,31 @@ int ESP8266Interface::connect(const char *ssid, const char *pass, nsapi_security
 
 int ESP8266Interface::connect()
 {
-    _esp.setTimeout(ESP8266_CONNECT_TIMEOUT);
-    
-    if (!_esp.reset()) {
-        return NSAPI_ERROR_DEVICE_ERROR;
-    }   
- 
+    size_t ssid_length = strlen(ap_ssid);
+    size_t pass_length = strlen(ap_pass);
+
+    if (ssid_length == 0 || ssid_length > ESP8266_SSID_MAX_LENGTH) {
+        return NSAPI_ERROR_PARAMETER;
+    }
+
+    if (ap_sec != NSAPI_SECURITY_NONE) {
+        if (pass_length < ESP8266_PASSPHRASE_MIN_LENGTH || pass_length > ESP8266_PASSPHRASE_MAX_LENGTH ) {
+            return NSAPI_ERROR_PARAMETER;
+        }
+    }
+
     _esp.setTimeout(ESP8266_MISC_TIMEOUT);
     
     if (_esp.get_firmware_version() != ESP8266_VERSION) {
         debug("ESP8266: ERROR: Firmware incompatible with this driver.\
                \r\nUpdate to v%d - https://developer.mbed.org/teams/ESP8266/wiki/Firmware-Update\r\n",ESP8266_VERSION); 
         return NSAPI_ERROR_DEVICE_ERROR;
     }
-    
+
+    if(_esp.isConnected()) {
```

Comment:
Should this be checked first when connecting?

Reply:
The order is same is in connect()

Reply:
I mean that since this is the connect() -function, shouldn't the isConnected be checked in the beginning of the function before starting to check the parameters?

Reply:
Ok, I'll change the order.